src/clib/xtg/grdcp3d_get_vtk_esg_geometry_data.c

```diff
@@ -0,0 +1,327 @@
+/*
+****************************************************************************************
+*
+* NAME:
+*    grdcp3d_get_vtk_esg_geometry_data.c
+*
+* DESCRIPTION:
+*    Get geometry data that is suitable as input to VTK's vtkExplicitStructuredGrid.
+*    See: https://vtk.org/doc/nightly/html/classvtkExplicitStructuredGrid.html#details
+*
+*    Basically what we have to do is build an unstructured grid representation where
+*    all grid cells are represented as hexahedrons with explicit connectivities.
+*    The connectivity table refers into the accompanying vertex table.
+*
+*    In VTK, cells order increases in I fastest, then J, then K.
+*
+*    This function also tries to remove/weld duplicate vertices, but this is a WIP.
+*
+* ARGUMENTS:
+*    ncol, nrow, nlay     i     Dimensions
+*    coordsv              i     Coordinate vector (xtgeo fmt)
+*    zcornsv              i     ZCORN vector (xtgeo fmt)
+*    vertArr              o     Will be populated with vertex coordinates (XYZ, XYZ, etc)
+*    connArr              o     Will be populated with the cell connectivities
+*    vertexCount_out      o     The actual number of XYZ vertices written into vertArr
+*
+* RETURNS:
+*    void, updates output arrays and argument
+*
+* TODO/ISSUES/BUGS:
+*    - verify/enhance removal of duplicate nodes
+*    - optimize the way cell corners are accessed
+*    - reduce memory usage in GeoMaker helper class
+*
+* LICENCE:
+*    CF XTGeo's LICENSE
+***************************************************************************************
+*/
+
+#include "libxtg.h"
+#include "libxtg_.h"
+#include "logger.h"
+
+
+// Forward declarations of static helpers
+static struct GeoMaker* gm_create(long cellCountI, long cellCountJ, long cellCountK, double* targetVertexBuffer, long* targetConnBuffer);
+static void gm_destroy(struct GeoMaker* gm);
+static void gm_addVertex(struct GeoMaker* gm, long i, long j, long k, double v[3]);
+static long gm_vertexCount(const struct GeoMaker* gm);
+static long gm_connectivityCount(const struct GeoMaker* gm);
+
+static double getTimeStamp_s();
+
+
+void grdcp3d_get_vtk_esg_geometry_data(long ncol,
+                                       long nrow,
+                                       long nlay,
+
+                                       double *coordsv,
+                                       long ncoordin,
+                                       float *zcornsv,
+                                       long nlaycornin,
+
+                                       double *vertArr,
+                                       long nVertArr,
+                                       long* connArr,
+                                       long nConnArr,
```

Comment:
We generally prefer `snake_case` style over `camelCase` in C code in xtgeo, if possible.

Reply:
Update public signature of `grdcp3d_get_vtk_esg_geometry()` to use snake_casing.
Is it worth the effort to rename all the internal variables used in the implementation file?

Reply:
Not a big deal, so leave it

Reply:
👍